L1Trigger/TrackerTFP/src/State.cc

```diff
@@ -0,0 +1,139 @@
+#include "L1Trigger/TrackerTFP/interface/State.h"
+
+using namespace std;
+using namespace trackerDTC;
+
+namespace trackerTFP {
+
+  //
+  State::State(State* state) :
+    dataFormats_(state->dataFormats_),
+    setup_(state->setup_),
+    track_(state->track_),
+    trackId_(state->trackId_),
+    parent_(state->parent_),
+    stub_(state->stub_),
+    layerMap_(state->layerMap_),
+    hitPattern_(state->hitPattern_),
+    x0_(state->x0_),
+    x1_(state->x1_),
+    x2_(state->x2_),
+    x3_(state->x3_),
+    C00_(state->C00_),
+    C01_(state->C01_),
+    C11_(state->C11_),
+    C22_(state->C22_),
+    C23_(state->C23_),
+    C33_(state->C33_),
+    numSkippedLayers_(state->numSkippedLayers_),
+    numConsistentLayers_(state->numConsistentLayers_)
+  {}
+
+  // proto state constructor
+  State::State(const DataFormats* dataFormats, TrackKFin* track, int trackId) :
+    dataFormats_(dataFormats),
+    setup_(dataFormats->setup()),
+    track_(track),
+    trackId_(trackId),
+    parent_(nullptr),
+    stub_(nullptr),
+    layerMap_(setup_->numLayers()),
```

Comment:
What should layerMap_ be initialized to to indicate that no stubs have yet been used in any layer?

Reply:
with setup_->numLayers()

Reply:
setup_numLayers() sets the number of elements in vector layerMap_, but it doesn't initialize the values of these elements.

Reply:
it should default initialize them

Reply:
https://en.cppreference.com/w/cpp/container/vector/vector states: